tests/test_unittest/test_gtc/test_oir.py

```diff
@@ -0,0 +1,39 @@
+import pytest
+from pydantic.error_wrappers import ValidationError
+
+from gt4py.gtc.common import DataType, ExprKind
+from gt4py.gtc.oir import Expr, HorizontalExecution
+
+
+A_ARITHMETIC_TYPE = DataType.INT32
+
+
+class DummyExpr(Expr):
+    """Fake expression for cases where a concrete expression is not needed."""
+
+    kind: ExprKind = ExprKind.FIELD
+
+
+def test_dtype_required():
+    with pytest.raises(ValidationError):
+        DummyExpr(dtype=None)
+
+
+@pytest.mark.parametrize(
+    "invalid_node,expected_regex",
+    [
+        (
+            lambda: HorizontalExecution(body=[], mask=DummyExpr(dtype=A_ARITHMETIC_TYPE)),
+            r".*must be.* bool.*",
+        ),
+        (
+            lambda: HorizontalExecution(
+                body=[], mask=DummyExpr(dtype=DataType.BOOL, kind=ExprKind.SCALAR)
+            ),
+            r".*must be.* field.*",
+        ),
+    ],
```

Comment:
I think `oir.AssignStatement` has a validator that isn't tested yet.

Reply:
done